### PR TITLE
feat(syntax-highlight): do client side

### PR DESCRIPTION
**Pull request by LeoMcA**
### Problem

Syntax highlighting in the content build system clutters our API, and would require porting to rari

### Solution

Do it client side

Also fixes e.g. vue pages having examples with a `MARKUP` header

---

## How did you test this change?

- `yarn dev`
- visited :3000 and :5042
- visited pages with various syntaxes: both ones which require dynamic importing, and those which are built into prism
- visited pages with examples with aliased and ignored syntaxes
- visited pages with markdown code blocks and manually crafted `<pre>`s
- tested streaming responses and loading responses from history in ai help

### CODE REVIEW

**Review by argl — APPROVED** [review]
Nice!

**Review by fiji-flo — APPROVED** [review]
Great. Let's see how it performs in prod.

**Review by caugner — COMMENTED** [review]
Overall LGTM, left some comments.

File: client/src/document/code/syntax-highlight.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+const PRISM_LANGUAGES = components.languages as Record<
+  string,
+  {
+    alias?: string | string[];
+    require?: string | string[];
+    optional?: string | string[];
+    [key: string]: any;
+  }
+>;
```
[the remark this answers is not in the document]
Comment (LeoMcA) [reply client/src/document/code/syntax-highlight.tsx:15]: No idea, guess whoever implemented it didn't feel like it?

**Review by caugner — APPROVED** [review]

**Pull request merged**